### PR TITLE
Publish reproducible evidence cases

### PR DESCRIPTION
**Pull request by bomly-guy**
## Summary

- add a versioned public evidence catalog with pinned inputs, checksums, exact reproduction commands, claim scope, and limitations
- add `make evidence` to validate the catalog or inspect one focused case
- publish plain-language case studies for dependency graphs, policy, reachability, read-only remediation guidance, scan targets, SBOM interoperability, portable unit-test/build assurance, and repeated performance measurements
- link the evidence guide from the main README and documentation navigation

The four commits keep the catalog, graph evidence, policy and guidance evidence, and target and operational evidence independently reviewable.

## User impact

Readers can now trace important behavior claims to public inputs and repeatable commands without private infrastructure. Live advisory observations, detector fallbacks, unknown evidence, analyzer tiers, mutable container tags, and operational limits are called out explicitly.

This PR changes repository assurance and documentation only. It does not change CLI behavior or the CLI, SDK, MCP, or plugin schemas.

## Validation

- `make evidence`
- focused catalog cases for graph, fallback, SPDX policy, persisted risk, and vulnerability policy
- pinned smoke cases for npm graph, SPDX ingestion, finding baselines, and source-change policy
- `make fmt-check`
- `make lint`
- `make test`
- `make build`
- `git diff --check origin/main...HEAD`

`make generate` was not required because no generated contract source changed.

## Summary by CodeRabbit

* **New Features**
  * Added a `make evidence` reproducible evidence command, including optional case filtering.
  * Introduced a new evidence level, **Snapshot**, for inputs whose results can move (e.g., container tags).

* **Documentation**
  * Added and linked new evidence documentation covering dependency-graph evidence, policy and guidance, and target/operation reproducibility details.
  * Updated public evidence catalog docs and manifests with the new level and reproducibility pinning.

* **Tests / Quality**
  * Strengthened public evidence validation and expanded coverage for edge cases (including path and symlink handling).

### CODE REVIEW

File: docs/manifest.json [hunk 1]
```diff
@@ -45,6 +45,12 @@
       "description": "Recipes for PR gates, SBOMs, triage, and license and offline scans.",
       "group": "start"
     },
+    {
+      "slug": "evidence",
```
Comment (bomly-guy) [inline docs/manifest.json:49]: Two thoughts on this entry:

1. `docs/evidence/` has three child pages, but this entry doesn't set `"hasChildren": true` — the other parents with nested directories (`detectors`, `matchers`, `schemas`, `plugins`, and `commands` in #350) all set it so the site nav exposes the children. Without it the case-study pages are reachable only through in-page links.
2. Is `start` the right group? This reads as assurance/reference material rather than a first-run step — `reference` ("Generated specifications and supported matrices") or even a case for `concepts` seems closer. Not blocking, but if you change it after #350 lands, move the `docs/README.md` bullet to the matching section in the same commit (the index mirrors the groups from then on).

File: internal/tools/publicevidence/main.go [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+func main() {
+	catalogPath := flag.String("catalog", defaultCatalog, "path to the public evidence catalog")
+	caseID := flag.String("case", "", "show one evidence case")
+	flag.Parse()
+
+	root, err := repositoryRoot()
+	if err != nil {
+		exitError(err)
+	}
+	loaded, err := loadCatalog(filepath.Join(root, filepath.FromSlash(*catalogPath)))
```
Comment (bomly-guy) [inline internal/tools/publicevidence/main.go]: Nit: an absolute `-catalog` path gets joined onto the repository root (`filepath.Join(root, "/abs/path")` → `root/abs/path`), which silently points at the wrong file. Worth honoring absolute paths:

```go
resolved := filepath.FromSlash(*catalogPath)
if !filepath.IsAbs(resolved) {
    resolved = filepath.Join(root, resolved)
}
```

Internal tool, so fine to defer.

**Review by bomly-guy — COMMENTED** [review]
Reviewed the catalog, the three case-study pages, the tool, and all 24 cases, and specifically checked interaction with the two open docs PRs (#350 wave 1, #354 wave 2, stacked). Overall this is strong work and looks ready: the catalog enforces exactly the right contract (sorted unique IDs, mandatory `proves` **and** `limitations`, checksum-verified artifacts, full 40-char revisions), reproduce commands delegate to the pinned smoke cases with `$`-anchored `-run` patterns per the slice convention, the prose never overclaims (live-service results framed as dated observations, mutable container tags called out, "unreachable does not mean safe"), and there are no competitor mentions. The tool itself is careful: traversal-guarded artifact paths, size-limited decode, `DisallowUnknownFields`, trailing-data rejection.

**Verified on a local three-way merge with the wave branches** (`docs/reference-wave1` + `docs/reference-wave2`):

- **No textual conflicts** — git auto-merges `README.md`, `docs/README.md`, `docs/manifest.json` cleanly in either merge order, and this branch already contains main@0f4b09b (#352).
- On the merged tree: `make evidence` passes (24 cases verified), the manifest sync test passes, and the offline link checker that #350 adds to CI passes over the new pages (744 links, 0 errors). So no coordination needed beyond one semantic point below.
- **One semantic note for whoever merges second:** #350 rewrites `docs/README.md` so its sections mirror the manifest groups. The auto-merge lands the Reproducible Evidence bullet in the "Getting started" section, which stays consistent with this PR's `"group": "start"` — but see the inline comment on whether `reference` is the better group; if the group changes, the bullet should move to the matching section in the same commit.

Two inline notes below (nav children flag, tool path nit). None of them block.
